enhancements/machine-config/install-time-support-for-image-mode-on-openshift.md

```diff
@@ -0,0 +1,342 @@
+
+We currently don't have support for Hypershift with post-install Image Mode. Install time support for this will be added (if needed) once we have post-install support for it. This is currently not part of the implementation plan for this enhancement. **SHOULD Hypershift support be in the Non-Goals section too?**
+
+#### Standalone Clusters
+
+Clusters in disconnected environments require special environment configurations such as a local package mirror, local image registry, etc.
+Post-install Image Mode currently works for disconnected environments and should work in a similar manner for install time
+
+#### SNO or MicroShift
+
+Single-Node OpenShift (SNO) currently works fine with post-install Image Mode and should continue to work as-is for install time support as well. However, there may be some efficiency improvements that can be made. For example, the final image could be written directly to the nodes’ filesystem before the handoff to rpm-ostree / bootc to avoid a registry roundtrip or a minimal registry should be started on the bootstrap node to host the built image. These efficiency gains would be best handled as a separate enhancement.
+
+### API Extensions
+
+#### Install Config Extensions
+
+```yaml
+# install-config.yaml
+apiVersion: v1
+kind: InstallConfig
+metadata:
+  name: my-cluster
+platform:
+  # ... existing platform config
+imageMode:
+  enabled: true
+  # Future: additional image mode configurations
+```
+
+#### MachineOSConfig (MOSC) Specifications
+
+MOSC YAML files for both `master` and `worker` MachineConfigPools will be placed in the installation directory and parsed by the bootstrap MCO:
+
+```yaml
+# mosc-master.yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineOSConfig
+metadata:
+  name: master
+spec:
+  machineConfigPool:
+    name: master
+  containerFile:
+  - content: |
+      FROM configs AS final
+      RUN dnf install -y specialized-driver
+      # Additional customizations
+  imageBuilder:
+    imageBuilderType: Job
+  renderedImagePushSpec: registry.example.com/custom-rhcos-master:latest
+  renderedImagePushSecret:
+    name: push-secret
+```
+
+```yaml
+# mosc-worker.yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineOSConfig
+metadata:
+  name: worker
+spec:
+  machineConfigPool:
+    name: worker
+  containerFile:
+  - content: |
+      FROM configs AS final
+      RUN dnf install -y special-worker-drivers
+      # Additional customizations
+  imageBuilder:
+    imageBuilderType: Job
+  renderedImagePushSpec: registry.example.com/custom-rhcos-worker:latest
+  renderedImagePushSecret:
+    name: push-secret
+```
+
+We will use the same MachineOSConfig CRD definition as already used for post-install customizations. The API for this can be found [here](https://github.com/openshift/api/blob/master/machineconfiguration/v1/types_machineosconfig.go).
```

Comment:
Since we aren't changing the API here, maybe we should move this out of the `API Extensions` section?

Reply:
Moved to implementation details